src/systems/systems/src/pressurization/cabin_pressure_simulation.rs

```diff
@@ -0,0 +1,252 @@
+    // Atmospheric constants
+    const R: f64 = 287.058; // Specific gas constant for air - m2/s2/K
+    const GAMMA: f64 = 1.4; // Rate of specific heats for air
+    const G: f64 = 9.80665; // Gravity - m/s2
+    const T_0: f64 = 288.2; // ISA standard temperature - K
+
+    // Aircraft constants
+    const RHO: f64 = 1.225; // Cabin air density - Kg/m3
+    const AREA_LEAKAGE: f64 = 0.0003; // m2
+    const CABIN_VOLUME: f64 = 400.; // m3
+    const OFV_SIZE: f64 = 0.03; // m2
+    const SAFETY_VALVE_SIZE: f64 = 0.02; //m2
+
+    pub fn new() -> Self {
+        Self {
+            previous_exterior_pressure: [Pressure::new::<hectopascal>(1013.25); 20],
+            exterior_pressure: Pressure::new::<hectopascal>(1013.25),
+            outflow_valve_open_amount: Ratio::new::<percent>(100.),
+            safety_valve_open_amount: Ratio::new::<percent>(0.),
+            z_coefficient: 0.0011,
+            flow_coefficient: 1.,
+            cabin_flow_in: VolumeRate::new::<cubic_meter_per_second>(0.),
+            cabin_flow_out: VolumeRate::new::<cubic_meter_per_second>(0.),
+            cabin_vs: Velocity::new::<meter_per_second>(0.),
+            cabin_pressure: Pressure::new::<hectopascal>(1013.25),
+        }
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        outflow_valve_open_amount: Ratio,
+        safety_valve_open_amount: Ratio,
+        packs_are_on: bool,
+        lgciu_gear_compressed: bool,
+        simulation_is_ground: bool,
+        should_open_outflow_valve: bool,
+    ) {
+        self.exterior_pressure = self.exterior_pressure_low_pass_filter(context);
+        self.z_coefficient = self.calculate_z();
+        self.flow_coefficient = self.calculate_flow_coefficient(should_open_outflow_valve);
+        self.outflow_valve_open_amount = outflow_valve_open_amount;
+        self.safety_valve_open_amount = safety_valve_open_amount;
+        self.cabin_flow_in = self.calculate_cabin_flow_in(packs_are_on, context);
+        self.cabin_flow_out = self.calculate_cabin_flow_out();
+        self.cabin_vs = self.set_cabin_vs();
+        self.cabin_pressure =
+            self.calculate_cabin_pressure(context, lgciu_gear_compressed, simulation_is_ground);
+    }
+
+    fn exterior_pressure_low_pass_filter(&mut self, context: &UpdateContext) -> Pressure {
+        let mut previous_pressure_vector: Vec<Pressure> = self.previous_exterior_pressure.to_vec();
+        previous_pressure_vector.remove(0);
+        previous_pressure_vector.push(context.ambient_pressure());
+
+        self.previous_exterior_pressure = previous_pressure_vector
+            .try_into()
+            .unwrap_or_else(|_| [context.ambient_pressure(); 20]);
+
+        let pressure_sum: f64 = self
+            .previous_exterior_pressure
+            .iter()
+            .map(|x| x.get::<hectopascal>())
+            .sum();
+
+        Pressure::new::<hectopascal>(pressure_sum / 20.)
+    }
```

Comment:
Thanks for these suggestions, I've implemented all of them. I added the `average` function to the `shared` module, as I think it's quite neat and can be used by other systems as well in the future. For now I implemented it only on the `Pressure` `uom` type, but it can be easily implemented on `f64` (or any other type) if required.